elyra/pipeline/component_registry.py

```diff
@@ -79,40 +80,32 @@ def __init__(self, component_registry_location: str, parser: ComponentParser):
     def registry_location(self) -> str:
         return self._component_registry_location
 
-    def get_all_components(self) -> Tuple[List[Component], List[Dict]]:
+    def get_all_components(self) -> List[Component]:
         """
         Retrieve all components from the component registry
         """
-        components: List[Component] = list()
-
-        # Read registry to find list of components
-        component_entries, category_entries = self._read_component_registry()
-
-        for component_entry in component_entries:
-            # Parse component details and add to list
-            component = self._parser.parse(component_entry)
-            if component:
-                components.extend(component)
 
-        return components, category_entries
+        components: List[Component] = self._read_component_registry().get('components')
+        return components
 
-    def get_component(self, component_id) -> Component:
+    def get_component(self, component_id: str) -> Component:
```

Comment:
I still don't understand why this method can't just look up its `id` in the result of `get_all_components()`.  Are the components in the list returned from `get_all_components()` different than the component returned from this method?

Reply:
That would be much easier. We currently don't store the results of `get_all_components`, but obviously if we did, we could access it on a call to `get_component` (like how it's done in the cache). Should we go ahead and do that? I just wasn't sure if there were other potential implications of this option vs. opening/reading the registry again

Reply:
Ah, right.  So you're saving the parsing of the other components and `parse()` loads the component as part of its operation.  That makes sense.  Please leave as-is.